packages/clay-navigation/src/Vertical.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import ClayIcon from '@clayui/icon';
+import React from 'react';
+import {Nav} from './Nav';
+import {NavItem} from './NavItem';
+import {NavLink} from './NavLink';
+
+interface IItemProps {
+	active?: boolean;
+	onClick?: () => void;
+	href?: string;
+	label?: string;
+}
+
+interface IItem extends IItemProps {
+	collapsed?: boolean;
+	items?: Array<IItemProps>;
+}
+
+interface IProps {
+	activeLabel: string;
+	items: Array<IItem>;
+	large?: boolean;
+	spritemap?: string;
+}
+
+function renderItems(
+	items: Array<IItem>,
+	collapsedMap: any,
+	setCollapsedMap: any,
+	spritemap?: string,
+	level = 0
+) {
+	return items.map(({active, href, items: subItems, label}, i) => {
+		const key = `${level}${i}`;
```

Comment:
Shouldn't we have a separator here? Absurd example: `level` 1/`i` 11 and `level` 11/ `i` 1 produce the same key (111). May not actually have a collision in practice but in general, if you are making a compound key out of two numbers, you need a separator to isolate the components.

Reply:
Good catch. just pushed an update.